### ISSUE TITLE
source for ethernet 8086:3101 device

### ISSUE THREAD

**Issue by sacarde** [jksinton/intel-igc#3]
hi,
   can you help me to install dkms for use device intel ethernet [8086:3101]

thank you

p.s.
in -v 5.4.0-7642.46
I found only [8086:3100]

**Comment by jksinton** [comment]
It looks like support for 8086:3101 was added in Linux Kernel 5.8–5.14 and 5.15-rc+HEAD.

See https://cateee.net/lkddb/web-lkddb/IGC.html

**Comment by sacarde** [comment]
oh, thank you

is possible to build only that module than all kernel?

by dkms for example

**Comment by jksinton** [comment]
Yes, you can pull the source for the igc driver in the latest kernel, and then build the dkms from that source.

I wrote some instructions on how to do this for Ubuntu:

https://jksinton.com/linux/building-a-linux-kernel-driver-on-ubuntu-using-dkms/

What kernel version are you running now?

**Comment by sacarde** [comment]
5.4.0

thank you, I read your link

**Comment by jksinton** [comment]
You might consider upgrading to a 5.8+ kernel with the support already built in. You could try building the driver under 5.4. But sometimes there  are new kernel features in newer releases that won't let you build the driver on an older kernel.

In my case, I used a dkms installation to implement a fix in the driver before it was adopted by the Linux Kernel.

**Comment by sacarde** [comment]
what you would suggest: upgrade from ubuntu-mainline-kernel or upgrade igc module by dkms?

**Comment by jksinton** [comment]
I would suggest upgrading your kernel, but you probably don't need to use the mainline kernel. You can probably just upgrade Ubuntu. For Ubuntu 20.04, you probably need to enable the HWE stack. The latest HWE stack for 20.04.3 LTS release is now on 5.11. Otherwise, you can upgrade to a different version of Ubuntu, for example, 21.04 is on 5.11 as well.
